db/error_handler.h

```diff
@@ -59,9 +71,11 @@ class ErrorHandler {
     bool IsBGWorkStopped() {
       return !bg_error_.ok() &&
              (bg_error_.severity() >= Status::Severity::kHardError ||
-              !auto_recovery_);
+              !auto_recovery_ || soft_error_no_bg_work_);
     }
 
+    bool IsSoftErrorNoBGWork() { return soft_error_no_bg_work_; }
```

Comment:
IIUC, this is only to prevent a manual flush thread from waiting indefinitely while recovery is going on. Can we just use ```IsBGWorkStopped```?

Reply:
Oh, yes. once `!bg_error_.ok() && soft_error_no_bg_work_`, `IsBGWorkStopped` will return true. It is the same as just check `soft_error_no_bg_work_` since `bg_error_` must be **non_ok**.